client/packages/system/src/Immunisation/ImmunisationProgramDetailView/VaccineCourseCreateModal.tsx

```diff
@@ -30,21 +34,30 @@ export const VaccineCourseCreateModal: FC<
   } = useVaccineCourse(programId);
   const isInvalid = !draft.name.trim();
 
+  const onOk = async () => {
+    const result = await create();
+    if (!result) return;
+
+    const response = result.centralServer.vaccineCourse.insertVaccineCourse;
+    if (response.__typename !== 'VaccineCourseNode') return;
+
+    navigate(
+      RouteBuilder.create(AppRoute.Programs)
+        .addPart(AppRoute.ImmunisationPrograms)
+        .addPart(response.programId)
+        .addPart(response.id)
+        .build()
+    );
+  };
+
   useEffect(() => {
     updatePatch({ programId: programId });
-  }, [programId]);
+  }, [programId, updatePatch]);
```

Comment:
This is causing perpetual re-rendering, I added a quick console log to check - I guess because updatePatch is a new function every time it is returned from the useVaccineCourse...

Probably means updatePatch should be in a `useCallback`, though I also wonder if we could just remove this useEffect, and pass the programId into the `create` called in the `onOk`?

Reply:
I should have known that agreeing to that suggestion would cause a problem! they usually do..
thanks, will fixit